instance/api/instance.py

```diff
@@ -51,9 +52,13 @@ def provision(self, request, pk=None):
             return Response({'status': 'Instance is not ready for reprovisioning'},
                             status=status.HTTP_403_FORBIDDEN)
 
-        instance.set_to_branch_tip()
-        provision_instance(pk)
+        try:
+            instance.set_to_branch_tip()
+        except github.ObjectDoesNotExist:
+            return Response({'status': "Ref '{0}' not found".format(instance.branch_name)},
+                            status=status.HTTP_403_FORBIDDEN)
```

Comment:
Not sure that 403 is the best status code to return here. I picked it because it's the one used above.